src/modules/schema/mim.object.schema.json

```diff
@@ -65,6 +65,9 @@
           },
           {
             "$ref": "#/definitions/integerMap"
+          },
+          {
+            "type": "null"

```

Comment:
This is invalid schema.

MIM does not allow for null object payloads. 

When there is no data to be reported, the module must fill in with blanks or default values and report that (values of  0 and empty strings, etc, or else) using correct object schema.

This PR cannot be accepted.

Reply:
https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins#back-end-operations
clearly states that desired properties must be removed by setting them to 'null'.

Our module cannot respond to 'null' values because the payload never reaches the module, the payload gets discarded before that. It cannot fill blanks in any way.

Reply:
Here's where the code aborts the processing of the payload: https://github.com/Azure/azure-osconfig/blob/4bd316b58e094cc635c29437c1a6c6fb88dc3f4e/src/agents/pnp/MpiProxy.c#L65
And here are the verbose logs for completeness

```
[2022-03-17 05:45:43] [PnpUtils.c:386] ModuleTwinCallback: received {"PackageManagerConfiguration":{"DesiredState":{"Packages":["cowsay"],"OldProperty":null}},"$version":34} (105 bytes)
[2022-03-17 05:45:43] [PnpUtils.c:335] Queued desired payload of 105 bytes at slot 3
[2022-03-17 05:45:43] [PnpAgent.c:295] Processing desired twin updates
[2022-03-17 05:45:43] [PnpUtils.c:223] ProcessJsonFromTwin: DEVICE_TWIN_UPDATE_PARTIAL
[2022-03-17 05:45:43] [PnpUtils.c:142] PropertyUpdateFromIotHubCallback: invoking PackageManagerConfiguration for property DesiredState, version 34
[2022-03-17 05:45:43] [PnpUtils.c:647] PackageManagerConfiguration.DesiredState: received {"Packages":["cowsay"],"OldProperty":null} (42 bytes)
[2022-03-17 05:45:43] [CommonUtils.cpp:242] [ERROR] MIM object JSON payload is invalid according to the schema
[2022-03-17 05:45:43] [CommonUtils.cpp:249] [ERROR] Invalid JSON: '{"Packages":["cowsay"],"OldProperty":null}' (42 bytes)
[2022-03-17 05:45:43] [MpiProxy.c:77] MpiSet(0xaaab1511e690, PackageManagerConfiguration, DesiredState, {"Packages":["cowsay"],"OldProperty":null}, 42 bytes) returned 22
[2022-03-17 05:45:43] [PnpUtils.c:660] [ERROR] PackageManagerConfiguration.DesiredState: MpiSet failed with 22
```

Reply:
Please read my comments again, I've been talking of **desired** properties.

Reply:
We will instruct the customer to modify the payload they send since setting a field within a MIM-object to null is not supported.